joomla/joomla-cms
# Solve word count bug in mod_feed + some code review

## About this PR

This PR solves a bug with the word count parameter of mod_feed. If you load a feed with HTML tags (for example: http://community.joomla.org/blogs/community.feed?type=rss) and set the word count to 20 there are way less than 20 words visible. The two things that cause this problem:
- `JHtml::_('string.truncate')` counts the characters, not the words of the string
- HTML tags like `<a href="http://link.nl">` are counted too

This PR change the text of the parameter to `Character count` and I also adds `strip_tags()` for the character count.

Other changes in this PR are code style and some change to make the code more readably and logic.
## Test instructions
1. Make a mod_feed module and use http://community.joomla.org/blogs/community.feed?type=rss as Feed URL, set the RTL Feed paramter on "No" and all other parameters on "Yes" for now
2. Set the "Word count" parameter on 20, and confirm the feed displays less than 20 words on the front-end
3. Apply the patch, and confirm the parameter is now called "Character limit" and the amount of characters is better now (Maybe still not 20 characters, because the function don't cut words in half)
4. Confirm all parameters are working like they use to do
5. Install an RTL language (Persion for example), and test the "RTL Feed" parameter for this language. If "RTL Feed" is set to no, even on RTL languages the feed must be displayed LTR
